packages/ember-routing/lib/system/route.js

```diff
@@ -1067,7 +1067,10 @@ let Route = EmberObject.extend(ActionHandler, Evented, {
   */
   transitionTo(name, context) {
     let router = this.router;
-    return router.transitionTo(...prefixRouteNameArg(this, arguments));
+    if (router.router.oldState) {
+      return router.transitionTo(...prefixRouteNameArg(this, arguments));
+    }
+    return router.replaceWith(...prefixRouteNameArg(this, arguments));
```

Comment:
I would prefer the logic be swapped around here so that the "default" thing we do is `transitionTo` from `Route#transitionTo`.